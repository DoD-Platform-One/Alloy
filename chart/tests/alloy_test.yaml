suite: Test Alloy statefulset 
templates:
  - controllers/statefulset.yaml
tests:
  - it: should match snapshot
    asserts:
      - matchSnapshot: {}
  - it: should set imagePullPolicy to Always
    set:
      image.pullPolicy: Always
    asserts:
      - equal:
          path: spec.template.spec.containers[0].imagePullPolicy
          value: Always
  - it: should imagePullPolicy to be default value IfNotPresent
    asserts:
      - equal:
          path: spec.template.spec.containers[0].imagePullPolicy
          value: IfNotPresent
  - it: should override securityContext
    set:
      podSecurityContext:
        runAsUser: 473
      alloy:
        securityContext:
          runAsUser: 473
    asserts:
      - equal:
          path: spec.template.spec.securityContext
          value:
            runAsUser: 473
      - equal:
          path: spec.template.spec.containers[0].securityContext
          value:
            allowPrivilegeEscalation: false
            capabilities:
              drop:
                - ALL
            readOnlyRootFilesystem: true
            runAsNonRoot: true
            runAsUser: 473
            seccompProfile:
              type: RuntimeDefault
  - it: should override hostNetwork
    set:
      controller:
        hostNetwork: true
    asserts:
      - equal:
          path: spec.template.controller.spec.hostNetwork
          value: true

