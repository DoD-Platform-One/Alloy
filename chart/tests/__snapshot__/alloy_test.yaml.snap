should match snapshot of default values:
  1: |
    apiVersion: apps/v1
    kind: StatefulSet
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: alloy
      name: alloy
      namespace: NAMESPACE
    spec:
      replicas: 1
      revisionHistoryLimit: 10
      selector:
        matchLabels:
          app.kubernetes.io/instance: alloy
          app.kubernetes.io/name: alloy
    template:
      metadata:
        creationTimestamp: null
        labels:
          app: alloy
          app.kubernetes.io/instance: alloy
          app.kubernetes.io/name: alloy
          version: v1.2.1
        annotations:
          kubectl.kubernetes.io/default-container: alloy
        spec:
          volumes:
            - name: config
              configMap:
                name: monitoring-alloy
                defaultMode: 420
          containers:
            - name: alloy
              image: registry1.dso.mil/ironbank/opensource/grafana/alloy:v1.2.1
              args:
                - run
                - /etc/alloy/config.alloy
                - '--storage.path=/tmp/alloy'
                - '--server.http.listen-addr=0.0.0.0:12345'
                - '--server.http.ui-path-prefix=/'
                - '--cluster.enabled=true'
                - '--cluster.join-addresses=monitoring-alloy-cluster'
                - '--stability.level=generally-available'
              ports:
                - name: http-metrics
                  containerPort: 12345
                  protocol: TCP
              env:
                - name: ALLOY_DEPLOY_MODE
                  value: helm
                - name: HOSTNAME
                  valueFrom:
                    fieldRef:
                      apiVersion: v1
                      fieldPath: spec.nodeName
              resources: {}
              volumeMounts:
                - name: config
                  mountPath: /etc/alloy
              readinessProbe:
                httpGet:
                  path: /-/ready
                  port: 12345
                  scheme: HTTP
                initialDelaySeconds: 10
                timeoutSeconds: 1
                periodSeconds: 10
                successThreshold: 1
                failureThreshold: 3
              terminationMessagePath: /dev/termination-log
              terminationMessagePolicy: File
              imagePullPolicy: IfNotPresent
              securityContext:
                capabilities:
                  drop:
                    - ALL
                runAsUser: 473
                runAsGroup: 473
                runAsNonRoot: true
          dnsPolicy: ClusterFirst
          hostNetwork: false
          serviceAccountName: monitoring-alloy
